layers.go

```diff
@@ -602,6 +606,66 @@ func (r *layerStore) Status() ([][2]string, error) {
 	return r.driver.Status(), nil
 }
 
+func (r *layerStore) PutFromAdditionalLayerStore(id string, parentLayer *Layer, d digest.Digest, imageref string) (layer *Layer, err error) {
+	adriver, ok := r.driver.(drivers.AdditionalLayerStoreDriver)
+	if !ok {
+		return nil, ErrLayerUnknown
+	}
+
+	var diffDir string
+	var layerInfo *Layer
+	for _, ls := range adriver.AdditionalLayerStores() {
+		if layerInfo, diffDir, err = searchLayer(ls, d, imageref); err == nil {
+			break
+		} else if err != ErrLayerUnknown {
+			return nil, err
+		}
+	}
+
```

Comment:
The code can get here without setting `layerInfo` and `diffDir`, e.g. if there are no ALS.

Reply:
Added a check to avoid this case.

Reply:
It can still happen if `searchLayer` never finds anything.

Reply:
Moved the logic into overlay driver (https://github.com/containers/storage/blob/05a4820e7372d0faba1d2c09ef78fd1226b495f4/drivers/overlay/overlay.go#L1490-L1513) and fixed this.